### PR TITLE
Refactor some GFX10 unified buffer format code

### PR DESCRIPTION
**Pull request by jayfoad**
Make it clearer which values are GFX10-specific. Use unsigned char for
arrays of format values just to save some space. Use 2D arrays when it
makes sense. Fix some member variable names. No change in behaviour.

### CODE REVIEW

**Comment by JaxLinAMD** [comment]
please ignore jenkins failures. see #1138

**Comment by amdvlk-admin** [comment]
### Test summary for commit 54c364c0e284db34922b6bcf8dec0f047f98d5ec
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 1e61f9ab6eb2041c73d7e7bd8f2021fa05a4bdd8</li>
<li>SPVGEN: dev 7c1cf34dd7eb4d3a6e6652c26292eef1bfcbcaf0</li>
<li>XGL: dev 2e6b9ee239ce2435550441fe08dd511ed0c0642f</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev bcd2fe473ee79c4907f2136ab963c1a3676c9db5</li>
</ul></details>

<details><summary>CTS tests (Failed: 336/224701)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26763/56156 (47.7%)</li>
<li><details><summary>Failed: 84/56156 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29309/56156 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26763/56156 (47.7%)</li>
<li><details><summary>Failed: 84/56156 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29309/56156 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25352/56233 (45.1%)</li>
<li><details><summary>Failed: 84/56233 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 30797/56233 (54.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26778/56156 (47.7%)</li>
<li><details><summary>Failed: 84/56156 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29294/56156 (52.2%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchInOutImportExport.cpp [hunk 1]
```diff
@@ -3465,15 +3471,15 @@ void PatchInOutImportExport::createStreamOutBufferStoreFunction(Value *storeValu
 
   format = formatOprd.u32All;
```
Comment (piotrAMD) [inline lgc/patch/PatchInOutImportExport.cpp:3472]: Does it also make sense to refactor the preceding switch, where for gfx10 we needlessly compute formatOprd?

Comment (jayfoad) [reply lgc/patch/PatchInOutImportExport.cpp]: Yes. I've rewritten more of this function.

Comment (piotrAMD) [reply lgc/patch/PatchInOutImportExport.cpp]: Thanks.

**Comment by jayfoad** [comment]
I've started using `switch (m_gfxIp.major) { ... }` more often, in the hope that it will reduce merge conflicts as and when we add support for future architectures.

**Review by piotrAMD — APPROVED** [review]
LGTM. 
(Note the clang-formatter prefers a more compact layout of the format list).

**Comment by jayfoad** [comment]
> (Note the clang-formatter prefers a more compact layout of the format list).

Yes. I decided to ignore it because I think the existing layout is clearer. Is that OK?

**Comment by piotrAMD** [comment]
> > (Note the clang-formatter prefers a more compact layout of the format list).
> 
> Yes. I decided to ignore it because I think the existing layout is clearer. Is that OK?

Fine with me.

**Review by amdrexu — APPROVED** [review]

**Comment by kuhar** [comment]
@jayfoad 
> Yes. I decided to ignore it because I think the existing layout is clearer. Is that OK?
Can you wrap it with `clang-format off` instead? https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code

**Comment by jayfoad** [comment]
> @jayfoad
> 
> > Yes. I decided to ignore it because I think the existing layout is clearer. Is that OK?
> 
> Can you wrap it with `clang-format off` instead? https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code

Good idea. Done.

**Comment by JaxLinAMD** [comment]
retest this please

**Pull request merged**